### DISCUSSION TITLE
Some considerations

### DISCUSSION THREAD

**Discussion by sgob1** [rhinstaller/anaconda-webui#37] [General]
Hello to all developers, testers and Fedora People,

I tested on GNOME Boxes the new Anaconda installer user interface, from
webui_latest_install ISO as in the Fedora Magazine article
instructions.

The installer interface is amazing: really polished, good-looking and
functional. Even though the previous interface was simple enough, this
one is even more welcoming, and I really enjoy the fact that now the
installation process is completely linear, one page after the other,
and there is no main page from which the user is expected to navigate
to other sections. I also love how the installation progress is
performed, which shows actual steps instead of a unique, long progress
bar. The new installer is really great!

Still, I have a few observations to make regarding the overall
experience. Please, don't take the following as criticisms, since I am
just a regular user which tried the new installer for the first time.

1. In the first page, Anaconda correctly identifies Italian as my
language, at least in GNOME Boxes. However, when reselecting the same
language again, the entire user interface completely reloads, leaving
the screen completely blank for a fraction of time. I would simply
expect the language to change without reloading the page, even with a
small, noticeable, delay.
2. Moreover, in the same page I would also add the usual 'test the
keyboard layout' box in which one can type its own accent characters,
such as those required to write in Italian, and similar glyphs.
3. In the second page, I see no options to specify custom disks and
partitions layout. In particular, I would like to see a page dedicated
to custom layouts, that can of course be skipped if an automatic
partitioning is specified, such as in the previous Anaconda.
4. Before installing, the previous Anaconda showed a list of the
partitions and disks that were going to be erased and created during
the installation. The new version only shows a warning that the
selected disks will be erased: it would be great if a 'show more'
button was available for advanced user to see what's going to happen.

The overall process was still buttery smooth, with the exception of the
'reboot' phase that hanged with a '[terminated]' message for an
unspecified amount of minutes.

**Comment by jkonecny12** [comment]
Hi @sgob1 thank you a lot for your feedback!

 
> Still, I have a few observations to make regarding the overall experience. Please, don't take the following as criticisms, since I am just a regular user which tried the new installer for the first time.
> 
> 1. In the first page, Anaconda correctly identifies Italian as my
>    language, at least in GNOME Boxes. However, when reselecting the same
>    language again, the entire user interface completely reloads, leaving
>    the screen completely blank for a fraction of time. I would simply
>    expect the language to change without reloading the page, even with a
>    small, noticeable, delay.

We will look on that issue.

> 2. Moreover, in the same page I would also add the usual 'test the
>    keyboard layout' box in which one can type its own accent characters,
>    such as those required to write in Italian, and similar glyphs.

Interesting idea, we will look onto that.

> 3. In the second page, I see no options to specify custom disks and
>    partitions layout. In particular, I would like to see a page dedicated
>    to custom layouts, that can of course be skipped if an automatic
>    partitioning is specified, such as in the previous Anaconda.

Yes, we are working on that, partitioning is a big topic. We are planning to get feedback from users specifically about the partitioning to decide about what partitioning method would be the best for users. Stay tuned :).

> 4. Before installing, the previous Anaconda showed a list of the
>    partitions and disks that were going to be erased and created during
>    the installation. The new version only shows a warning that the
>    selected disks will be erased: it would be great if a 'show more'
>    button was available for advanced user to see what's going to happen.

Thanks for the feedback we already get this feedback. It's great that we have a confirmation.
